### PR TITLE
Add help command

### PR DESCRIPTION
**Pull request by jsvana**
Allows users to see what a command does

### CODE REVIEW

File: src/client.rs [hunk 1]
```diff
@@ -128,6 +128,17 @@ impl Client {
             .into_inner())
     }
 
+    pub async fn registered_commands(&self) -> Result<CommandsResponse> {
+        Ok(self
+            .inner
+            .lock()
+            .await
+            .inner_mut_ref()
+            .registered_commands(CommandsRequest {})
```
Comment (guswynn) [inline src/client.rs:137]: does this talk to core?

File: src/plugins/help.rs [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+pub struct HelpPlugin {}
+
+impl HelpPlugin {
+    pub fn new() -> Self {
+        Self {}
+    }
+
+    async fn handle_help(&self, ctx: &Context, argument: Option<&str>) -> Result<()> {
+        let response = ctx.registered_commands().await?;
```
Comment (guswynn) [inline src/plugins/help.rs:11]: confusing var name

File: src/plugins/help.rs [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+    fn command_metadata(&self) -> Vec<CommandMetadata> {
+        vec![CommandMetadata {
+            name: "help".to_string(),
+            short_help: "gives usage for a command".to_string(),
+            full_help: "gives usage and other information for a command".to_string(),
+        }]
+    }
+
+    async fn run(self, bot: Arc<Client>) -> Result<()> {
```
Comment (guswynn) [inline src/plugins/help.rs:54]: random q, why do we pass in a full Arc? that mean's each run requires a ref count bump? why not just &Client from an arc Deref?

Comment (belak) [reply src/plugins/help.rs:54]: I believe there are times when we want to start a background task from a plugin which requires an Arc. Also, this isn't as bad as you'd expect - it's one reference per plugin and should only happen when starting up or spawning  task.

File: src/client.rs (hunk 1 above)
Comment (belak) [reply src/client.rs:137]: Yep

**Pull request merged**